Sources/Core/TextEditor/View/UIKit/TextEditorUIView.swift

```diff
@@ -0,0 +1,412 @@
+        self.viewModel.$font.subscribe(in: &self.cancellables) { [weak self] font in
+            guard let self else { return }
+            self.font = font.uiFont
+            self.placeholderLabel.font = font.uiFont
+        }
+
+        self.viewModel.$shouldShowPlaceholder
+            .removeDuplicates().subscribe(in: &self.cancellables) { [weak self] shouldShowPlaceholder in
+                guard let self else { return }
+
+                self.placeholderLabel.isHidden = !shouldShowPlaceholder
+            }
+    }
+
+    // MARK: - Notification Center
+
+    private func setupNotificationCenter() {
+        for name in self.notificationNames {
+            NotificationCenter.default.addObserver(
+                self,
+                selector: #selector(textViewFired(notification:)),
+                name: name,
+                object: self
+            )
+        }
+    }
+
+    private func unsetNotificationCenter() {
+        for name in self.notificationNames {
+            NotificationCenter.default.removeObserver(
+                self,
+                name: name,
+                object: self
+            )
+        }
+    }
+
+    // MARK: - Update
+
+    private func updatePaddings(
+        leftSpacing: CGFloat? = nil,
+        rightSpacing: CGFloat? = nil
+    ) {
+        let leftSpacing = leftSpacing ?? self.viewModel.leftSpacing
+        let verticalSpacing = self.viewModel.getVerticalSpacing(from: self.scaleHeight)
+        let rightSpacing = rightSpacing ?? self.viewModel.rightSpacing
+
+        // Containerzaoipjoi
+        self.textContainerInset = UIEdgeInsets(
+            top: verticalSpacing,
+            left: leftSpacing,
+            bottom: verticalSpacing,
+            right: rightSpacing
+        )
+
+        // Placeholder
+        if let placeholderWidthConstraint {
+            placeholderWidthConstraint.constant = -(leftSpacing + rightSpacing)
+        }
+
+        for constraint in self.placeholderVerticalPaddingConstraints {
+            constraint.constant = verticalSpacing
+        }
+
+        self.placeholderLabel.sizeToFit()
+        self.placeholderLabel.updateConstraintsIfNeeded()
+    }
+
+    private func updateBorder() {
+        self.setCornerRadius(self.scaleCornerRadius)
+        self.setBorderWidth(self.scaleBorderWidth)
+    }
+
+    // MARK: - Action
+
+    @objc
+    private func textViewFired(notification: Notification) {
+        self.viewModel.contentChanged(with: self.text)
+    }
+
+    // MARK: - Responder
+
+    public override var canBecomeFirstResponder: Bool {
+        return self.isEnabled ? super.canBecomeFirstResponder : false
+    }
+
+    public override func becomeFirstResponder() -> Bool {
+        let bool = super.becomeFirstResponder()
+        self.viewModel.isFocused = bool
+        return bool
+    }
+
+    public override func resignFirstResponder() -> Bool {
+        super.resignFirstResponder()
+        self.viewModel.isFocused = false
+        return true
```

Comment:
public override func resignFirstResponder() -> Bool {
        self.viewModel.isFocused = false
        return super.resignFirstResponder()
}

Reply:
You're right ! TY